keras/backend/tensorflow_backend.py

```diff
@@ -11,7 +11,7 @@
 _SESSION = None
 _LEARNING_PHASE = tf.placeholder(dtype='uint8', name='keras_learning_phase')  # 0 = test, 1 = train
 _MANUAL_VAR_INIT = False
-
+_DEVICE = None
```

Comment:
I think that would be better indeed. It's not a big deal if the internal implementations differ as long as the external interface is the same.